caffe2-cnn/mnist_nn_cpu.py

```diff
@@ -0,0 +1,197 @@
+import csv
+import math
+import numpy as np
+import os
+import shutil
+import urllib2
+from caffe2.python import workspace, core, cnn, net_drawer, visualize
+from sigopt import Connection
+
+current_folder = os.path.join(os.path.expanduser('~'), 'caffe2_notebooks')
+
+data_folder = os.path.join(current_folder, 'tutorial_data', 'mnist')
+root_folder = os.path.join(current_folder, 'tutorial_files', 'tutorial_mnist')
+image_file_train = os.path.join(data_folder, "train-images-idx3-ubyte")
+label_file_train = os.path.join(data_folder, "train-labels-idx1-ubyte")
+image_file_test = os.path.join(data_folder, "t10k-images-idx3-ubyte")
+label_file_test = os.path.join(data_folder, "t10k-labels-idx1-ubyte")
+
+# Get the dataset if it is missing
+def download_dataset(url, path):
+  import requests, zipfile, StringIO
+  print "Downloading... ", url, " to ", path
+  r = requests.get(url, stream=True)
+  z = zipfile.ZipFile(StringIO.StringIO(r.content))
+  z.extractall(path)
+
+# Load training/testing data into db
+def generate_db(image, label, name):
+  name = os.path.join(data_folder, name)
+  print 'DB: ', name
+  if not os.path.exists(name):
+    syscall = "/usr/local/binaries/make_mnist_db --channel_first --db leveldb --image_file " + image + " --label_file " + label + " --output_file " + name
+    # print "Creating database with: ", syscall
```

Comment:
Yep can do, but I'll leave the print of best assignments since if they ran this out of the box they'd probably want to see some sort of output